datasets/seq_cifar100_label_drift.py

```diff
@@ -0,0 +1,237 @@
+        """
+        img, target = self.data[index], self.targets[index]
+
+        # doing this so that it is consistent with all other datasets
+        # to return a PIL Image
+        img = Image.fromarray(img)
+        img = self.transform(img)
+        target = self.superclass_mapping[target]
+        target = self.metaclass_mapping[target]
+
+        return img, target
+
+    def select_classes(self, classes_list: list[int]):
+        if len(classes_list) == 0:
+            self.data = np.array([])
+            self.targets = np.array([])
+            self.classes = []
+            return
+
+        mask = np.zeros_like(np.array(self.targets))
+        for label in classes_list:
+            mask = np.logical_or(mask, np.array(self.targets) == label)
+        self.data = self.data[mask]
+        self.targets = np.array(self.targets)[mask]
+
+        self.classes = classes_list
+
+    def apply_drift(self, classes: list):
+        if len(set(self.classes).union(classes)) == 0:
+            return
+        self.drifted_classes.extend(classes)
+
+        # switch meta-class in target_transform dictionary
+        for c in classes:
+            s = self.superclass_mapping[c]
+            self.metaclass_mapping[s] = 0 if self.metaclass_mapping[s] == 1 else 1
+
+    def prepare_normal_data(self):
+        pass
+
+
+class TargetTransform:
+    def __init__(self, superclass_mapping, metaclass_mapping):
+        self.superclass_mapping = superclass_mapping
+        self.metaclass_mapping = metaclass_mapping
+
+
+class SequentialCIFAR100LabelDrift(ContinualDataset):
+
+    NAME = "seq-cifar100-label-drift"
+    SETTING = "class-il"
+    N_CLASSES_PER_TASK = 5
+    N_TASKS = 20
+
+    TRANSFORM = transforms.Compose(
+        [
+            transforms.RandomCrop(32, padding=4),
+            transforms.RandomHorizontalFlip(),
+            transforms.ToTensor(),
+        ]
+    )
+
+    NO_AUG_TRANSFORM = transforms.Compose([transforms.ToTensor()])
+
+    TARGET_TRANSFORM = TargetTransform(
+        superclass_target_mapping, metaclass_target_mapping
+    )
+
+    def get_dataset(self, train=True):
+        """returns native version of represented dataset"""
+
+        if train:
+            return TrainCIFAR100LabelDrift(
+                base_path() + "CIFAR100LabelDrift",
+                transform=self.TRANSFORM,
+                target_transform=self.TARGET_TRANSFORM,
+                not_aug_transform=self.NO_AUG_TRANSFORM,
+            )
+        else:
+            return TestCIFAR100LabelDrift(
+                base_path() + "CIFAR100LabelDrift",
+                transform=self.NO_AUG_TRANSFORM,
+                target_transform=self.TARGET_TRANSFORM,
+            )
+
+    def get_transform(self):
+        transform = transforms.Compose([transforms.ToPILImage(), self.TRANSFORM])
+        return transform
+
+    @staticmethod
+    def get_backbone():
+        backbone = resnet18(nclasses=2)
+        backbone.conv1 = conv3x3(3, backbone.nf)
```

Comment:
This line is not needed. It was introduced in seq-fashionmnist to change number of kernels in the first layer (fashion-mnist has only 1 channel). You don't need it here. Standard resnet18 already has 3 kernels in the first layer, so there is no need for this line.